chore(#10016): implement version specific helm charts

lorerod opened the pull request:
<!--
Please use semantic PR titles that respect this format:

<type>(#<issue number>): <subject>

Quick example:

feat(#1234): add hat wobble
^--^(#^--^): ^------------^
|     |      |
|     |      + - > subject
|     |
|     + -------- > issue number
|
+ -------------- > type: chore, feat, fix, perf.

https://docs.communityhealthtoolkit.org/contribute/code/workflow/#commit-message-format
-->

# Description

closes #10016

<!-- ISSUE NUMBER -->

# Code review checklist
<!-- Remove or comment out any items that do not apply to this PR; in the remaining boxes, replace the [ ] with [x]. -->
- [ ] UI/UX backwards compatible: Test it works for the new design (enabled by default). And test it works in the old design, enable `can_view_old_navigation` permission to see the old design. Test it has appropriate design for RTL languages. 
- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.



# Compose URLs
If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:10016-version-specific-helm-charts/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:10016-version-specific-helm-charts/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:10016-version-specific-helm-charts/docker-compose/cht-couchdb-clustered.yml)


# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

lorerod commented:
_Note to myself:_ **Scenarios to test**
1. New, single, k3s-k3d
2. New, multi, k3s-k3d
3. Migration, single, k3s-k3d
4. Migration, multi, k3s-k3d
5. New, single, eks
6. New, multi, eks
7. Migration, single, eks
8. Migration, multi, eks
9. New, single, gke
10. New, multi, gke
11. Migration, single, gke
12. Migration, multi, gke

dianabarsan commented on scripts/build/helm/templates/sentinel/deployment.yaml:
```diff
@@ -0,0 +1,34 @@
+                  key: COUCH_URL
+            - name: API_PORT
+              value: '5988'
+          image: {{ .Values.upstream_servers.docker_registry }}/cht-sentinel:{{ .Values.cht_image_tag }}
```
Please quote this value.

dianabarsan commented on scripts/build/helm/templates/platforms/aws/persistent-volume.yaml:
```diff
@@ -0,0 +1,46 @@
+  name: couchdb-pv-{{ .Values.namespace }}
+spec:
+  capacity:
+    storage: {{ .Values.ebs.preExistingEBSVolumeSize | default "100Gi" }}  # Fallback for backward compatibility
```
I think we should just require specific values. we've already had one deployment ship with default values that failed due to disk space shortly after launch.

lorerod commented on scripts/build/helm/templates/couchdb/deployment.yaml:
```diff
@@ -0,0 +1,255 @@
+{{- /* Validation for required values */ -}}
+{{- if not .Values.project_name }}
```
I have created _validations.tpl

lorerod commented:
Hi @dianabarsan,
Just an update on this PR — I’ve now implemented your feedback and made additional progress:
- Integrated the new Helm charts with GitHub Actions to generate release assets
- Updated the integration tests to run using the new charts

One thing I wanted to flag: to get the integration tests to pass, I had to modify the deployment strategies for the API, CouchDB, and Sentinel. I'm unsure whether the strategy used in the original Helm charts was intentional due to specific constraints or deployment behavior. If you have any information on that, I’d appreciate your insight. 
cc: @Hareet 

If you have time for another round of review, I’d love your thoughts on the current state and any other adjustments you think should be made.

Thanks again!

dianabarsan reviewed (approved):
Outstanding job, @lorerod ! Massive improvements in readability, and I love that lo little changes were required for the CI tests to work with the new templates!
